mettle/worker/worker.go

```diff
@@ -915,6 +924,24 @@ func (w *worker) update(stage pb.ExecutionStage_Value, response *pb.ExecuteRespo
 	return common.PublishWithOrderingKey(ctx, w.responses, body, w.actionDigest.Hash, w.name)
 }
 
+// periodicallyPushMetrics will push this worker's metrics to the gateway every 5 minutes.
+func (w *worker) periodicallyPushMetrics() {
+	if w.promGatewayURL != "" {
+		// Set up the metrics
+		pusher := push.New(w.promGatewayURL, "mettle")
+		for _, metric := range metrics {
+			pusher = pusher.Collector(metric).Format(expfmt.FmtText)
+		}
+		// Push to the gateway every 5 minutes
+		for {
```

Comment:
One more thing - would be nicer to use a `time.Ticker` here rather than sleeping each time